File: backend/app/api/odds.py
"""Odds cache HTTP endpoint.

Plan §9: GET /api/odds returns the current shared odds cache. Unauthenticated
(matches the SvelteKit /odds endpoint's posture) — the SvelteKit /odds is now
a thin proxy that calls this endpoint and returns the result to the client.

The cache itself lives at /data/odds_cache.json. See
backend/app/services/odds_cache.py for the read-through semantics and the
durability contract.
"""

from fastapi import APIRouter

from app.services.odds_cache import get_or_refresh_odds, get_or_refresh_outrights

router = APIRouter()


@router.get("/")
async def get_odds() -> dict:
    """Return the current odds cache, refreshing if stale.

    Always returns HTTP 200. The body contains either:
      - {fetchedAt, remainingRequests, usedRequests, matches}  — success path
      - {error: 'not_configured', matches: []}                  — key unset
      - {error: 'fetch_failed', matches: []}                    — cold-start + upstream failure

    The frontend inspects the body to decide how to react. This 200-always
    posture mirrors the existing SvelteKit /odds endpoint and keeps client
    branching simple.
    """
    return await get_or_refresh_odds()


@router.get("/outrights")
async def get_outrights() -> dict:
    """Return the current tournament-winner odds cache, refreshing if stale.

    Always returns HTTP 200. The body contains either:
      - {fetchedAt, remainingRequests, usedRequests, teams}  — success path,
        `teams` is the top 10 by averaged de-vigged implied probability
      - {error: 'not_configured', teams: []}                   — key unset
      - {error: 'fetch_failed', teams: []}                     — cold-start + upstream failure
    """
    return await get_or_refresh_outrights()

File: frontend/src/lib/api/odds.ts
/**
 * Championship (tournament-winner) odds API — GET /api/odds/outrights.
 *
 * Separate from the Smart Fill h2h odds cache (`/odds` SvelteKit proxy +
 * `frontend/src/lib/utils/oddsCache.ts`): this hits the FastAPI cache
 * directly like every other v4 endpoint (`$api/leaderboard` etc.), no
 * SvelteKit proxy hop needed.
 */

import { api } from './client';

export interface ChampionshipOddsTeam {
	team: string;
	probability: number; // 0..1, de-vigged + averaged across bookmakers
}

export interface ChampionshipOddsResponse {
	fetchedAt?: string;
	teams: ChampionshipOddsTeam[];
	error?: 'not_configured' | 'fetch_failed';
}

export async function getChampionshipOdds(): Promise<ChampionshipOddsResponse> {
	return api.get<ChampionshipOddsResponse>('/odds/outrights');
}
